frontend/src/components/AddConcepts.tsx

```diff
@@ -0,0 +1,134 @@
+  {
+    selector: row => row.distinct_person_cnt,
+    name: "Distinct Person Count",
+    sortable: true,
+    width: 100,
+  },
+];
+
+export function AddConcepts() {
+  return <ConceptStringSearch />;
+
+}
+// from https://stackoverflow.com/a/66167322/1368860
+function ConceptStringSearch() {
+  const dataGetter: DataGetter = useDataGetter();
+  let concepts: Concept[] = [];
+  const [searchText, setSearchText] = React.useState("");
+  const [concept_ids, setConceptIds] = React.useState(concepts);
+  const lastRequest = React.useRef(null);
+
+  // this effect will be fired every time searchText changes
+  React.useEffect(() => {
+    (async () => {
+      // setting min lenght for searchText
+      if (searchText.length >= 3) {
+        // updating the ref variable with the current searchText
+        lastRequest.current = searchText;
+        const r = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concept_search, searchText)
+        if (lastRequest.current === searchText) {
+          console.log("response is valid!", r);
+          setConceptIds(r);
+        } else {
+          console.log("discarding api response", searchText, lastRequest.current);
+        }
+      }
+    })();
+  }, [searchText]);
+
+  return (
+    <div className="App" style={{paddingLeft: 100}}>
+      <h1>Concept Search</h1>
+      <input style={{width: 300}} type="text" placeholder="match characters in concept name"
+             onChange={(e) => setSearchText(e.target.value)}
+             value={searchText}
+             autoFocus={true}
+      />{'\u00A0'}{'\u00A0'}{'\u00A0'}{concept_ids.length ? concept_ids.length.toLocaleString() + ' concept_ids' : ""}
+      <ConceptTable concept_ids={concept_ids} />
+    </div>
+  );
+}
+
+function ConceptTable(props) {
+  let {concept_ids} = props;
+  const dataGetter = useDataGetter();
+  const [concepts, setConcepts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const totalRows = concept_ids.length;
+  const [perPage, setPerPage] = useState(10);
+
+  const fetchConcepts = async page => {
+    setLoading(true);
+    let ids = concept_ids.slice(page - 1, perPage);
+    let conceptLookup = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concepts, ids);
+    const _concepts = ids.map(id => conceptLookup[id]);
+    setConcepts(_concepts);
+    setLoading(false);
+  };
+  const handlePageChange = page => {
+    fetchConcepts(page);
+  };
+  const handlePerRowsChange = async (newPerPage, page) => {
+    setLoading(true);
+    let ids = concept_ids.slice(page - 1, perPage);
+    let conceptLookup = await dataGetter.fetchAndCacheItems(dataGetter.apiCalls.concepts, ids);
+    const _concepts = ids.map(id => conceptLookup[id]);
+    setConcepts(_concepts);
+    setPerPage(newPerPage);
+    setLoading(false);
+  };
+  useEffect(() => {
+    fetchConcepts(1); // fetch page 1 of users
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [concept_ids]);
+  console.log(concepts);
```

Comment:
Do we still need this, or was this for troubleshooting?

FYI, it's printing out a lot. It printed 10+ times just when I did my search for 'infarction', before I went through any pages.

<details><summary>Details</summary>
<p>

```
api_call_group_id 20417
MuiAppBar.jsx:95 null
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:72 discarding api response infarcti infarction
AddConcepts.tsx:72 discarding api response infarct infarction
AddConcepts.tsx:72 discarding api response infarc infarction
AddConcepts.tsx:72 discarding api response inf infarction
AddConcepts.tsx:72 discarding api response infar infarction
AddConcepts.tsx:72 discarding api response infa infarction
AddConcepts.tsx:72 discarding api response infarctio infarction
AddConcepts.tsx:69 response is valid! Array(1518)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(0)
AddConcepts.tsx:123 Array(10)
```

</p>
</details>